src/compute-client/src/plan/top_k.rs

```diff
@@ -171,9 +171,10 @@ pub struct MonotonicTop1Plan {
     pub group_key: Vec<usize>,
     /// Ordering that is used within each group.
     pub order_key: Vec<mz_expr::ColumnOrder>,
-    /// True if the input is logically but not physically monotonic,
-    /// and the operator must first consolidate the inputs to remove
-    /// potential negations.
+    /// True if the input is not physically monotonic, and the operator must perform
+    /// consolidation to remove potential negations. The operator implementation is
+    /// free to consolidate as late as possible while ensuring correctness, so it is
+    /// not a requirement that the input be directly subjected to consolidation.
```

Comment:
nit: maybe reference the design doc (I think you can edit all `pub must_consolidate: bool` docstrings at once by doing a global "find and replace" in your IDE)?

Reply:
Cool, links to the design doc added in the respective docstrings.